db/de.go

```diff
@@ -86,3 +87,74 @@ func (d *DEDatabase) AddUserDataUsage(context context.Context, username string,
 
 	return d.doUserUsage(context, query)
 }
+
+func (d *DEDatabase) AddUserDataUsageBatch(context context.Context, usages map[string]int64, time time.Time) ([]*UserDataUsage, error) {
+	log.Tracef("Inserting usages: %+v at %s", usages, time)
+	var placeholders []string
+	var startargs []interface{}
+	for usr, usg := range usages {
+		placeholders = append(placeholders, "(?::text, ?::bigint)")
+		startargs = append(startargs, usr, usg)
+	}
+	startcte := "WITH new_usages (username, usage) AS (VALUES " + strings.Join(placeholders, ",") + ")"
+
+	querys, args, err := psql.Insert(d.Table("user_data_usage", "d")).
+		Prefix(startcte, startargs...).
```

Comment:
It's annoying that squirrel doesn't support CTEs using `WITH` explicitly, but we can do them this way.